plugin-core/src/main/java/appland/config/AppMapConfigFile.java

```diff
@@ -31,11 +31,14 @@ public class AppMapConfigFile {
     private static final String NAME_PROPERTY = "name";
     private static final String APPMAP_DIR_PROPERTY = "appmap_dir";
     private static final String PACKAGES_PROPERTY = "packages";
+    private static final String LANGUAGE_PROPERTY = "language";
+    private static final String LANGUAGE_VALUE = "java";
 
     private final @NotNull ObjectNode yamlData;
 
     public AppMapConfigFile() {
         this(new ObjectNode(JsonNodeFactory.instance));
+        yamlData.set(LANGUAGE_PROPERTY, new TextNode(LANGUAGE_VALUE));
```

Comment:
For consistency, I'd have put this line into `AppMapJavaPackageConfig`, but if you don't want to allow a missing language by default, then the constructor here is probably the better location.
https://github.com/getappmap/appmap-intellij-plugin/blob/3ff6d6f98bc432e6846ecff32627d81b12f715db/plugin-java/src/main/java/appland/execution/AppMapJavaPackageConfig.java#L117-L121

Reply:
Yes, I put it here deliberately, because it should never be unset.